app/src/main/java/com/example/themealsapp/utils/BaseFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.themealsapp.utils
+
+import android.app.AlertDialog
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.ViewModelProvider
+import com.example.themealsapp.R
+import com.example.themealsapp.presentation.viewmodel.MealsViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+open class BaseFragment : Fragment() {
+
+    protected val mealsViewModel: MealsViewModel by lazy {
+        ViewModelProvider(requireActivity())[MealsViewModel::class.java]
```

Comment:
whats the difference between require activity and activity variable?

Reply:
It refers to the viewModelOwner. It will return the Activity where this fragment is associated with. It will not let you create the ViewModel if you pass Activity().